install-jellyfin.ps1

```diff
@@ -0,0 +1,385 @@
+[CmdletBinding()]
+
+param(
+    [Switch]$Quiet,
+    [Switch]$InstallAsService,
+    [pscredential]$ServiceUser,
+    [switch]$CreateDesktopShorcut,
+    [switch]$LaunchJellyfin,
+    [switch]$MigrateEmbyLibrary,
+    [string]$InstallLocation,
+    [string]$EmbyLibraryLocation
+)
+<# This form was created using POSHGUI.com  a free online gui designer for PowerShell
+.NAME
+    Install-Jellyfin
+#>
+# Get the ID and security principal of the current user account
+$myWindowsID=[System.Security.Principal.WindowsIdentity]::GetCurrent()
+$myWindowsPrincipal=new-object System.Security.Principal.WindowsPrincipal($myWindowsID)
+ 
+# Get the security principal for the Administrator role
+$adminRole=[System.Security.Principal.WindowsBuiltInRole]::Administrator
+ 
+# Check to see if we are currently running "as Administrator"
+if ($myWindowsPrincipal.IsInRole($adminRole))
+   {
+   # We are running "as Administrator" - so change the title and background color to indicate this
+   $Host.UI.RawUI.WindowTitle = $myInvocation.MyCommand.Definition + "(Elevated)"
+   $Host.UI.RawUI.BackgroundColor = "DarkBlue"
+   clear-host
+   }
+else
+   {
+   # We are not running "as Administrator" - so relaunch as administrator
+   
+   # Create a new process object that starts PowerShell
+   $newProcess = new-object System.Diagnostics.ProcessStartInfo "PowerShell";
+   
+   # Specify the current script path and name as a parameter
+   $newProcess.Arguments = $myInvocation.MyCommand.Definition;
+   
+   # Indicate that the process should be elevated
+   $newProcess.Verb = "runas";
+   
+   # Start the new process
+   [System.Diagnostics.Process]::Start($newProcess);
+   
+   # Exit from the current, unelevated, process
+   exit
+   }
+
+if($Quiet.IsPresent -or $Quiet -eq $true){
+    $Script:JellyfinDataDir = "$env:AppData\jellyfin\"
+    if([string]::IsNullOrEmpty($InstallLocation)){$Script:DefaultJellyFinInstallDirectory = "$env:Appdata\jellyfin\"}else{$Script:DefaultJellyFinInstallDirectory = $InstallLocation}
+    
+    if([string]::IsNullOrEmpty($EmbyLibraryLocation)){$Script:defaultEmbyDataDir = "$env:Appdata\Emby-Server\data\"}else{$Script:defaultEmbyDataDir = $EmbyLibraryLocation}
+    
+    if($InstallAsService.IsPresent -or $InstallAsService -eq $true){$Script:InstallAsService = $true ; $Script:JellyfinDataDir = "C:\WINDOWS\system32\config\systemprofile\AppData\Roaming\jellyfin\" }else{$Script:InstallAsService = $false}
```

Comment:
Might look prettier if not written in one line

Reply:
fixed in latest